addons/account/models/account_bank_statement.py

```diff
@@ -815,10 +570,20 @@ def create(self, vals_list):
 
             # Otherwise field narration will be recomputed silently (at next flush) when writing on partner_id
             self.env.remove_to_compute(st_line.move_id._fields['narration'], st_line.move_id)
+
+        # No need for the user to manage their status (from Draft to Posted)
+        st_lines.move_id.action_post()
+        for st_line in st_lines.filtered('statement_id.attachment_ids'):
+            st_line._post_attachments_to_move()
         return st_lines
 
     def write(self, vals):
         # OVERRIDE
+        if 'date' in vals:
+            # prevents error when trying to change the date of a bank statement line and the sequence of the related
+            # move does not match the date (the date is in a different period)
+            self.move_id.name = '/'
```

Comment:
This diff is scarying me... :scream:

Reply:
as discussed: this shouldn't be handled in this task, or specifically on the bank statement lines.

https://www.odoo.com/web#id=2976499&cids=1&menu_id=4720&action=333&active_id=967&model=project.task&view_type=form

Reply:
Is there any alternative, (other than erasing a move and recreate it, make a move draft and repost it)? The last discussion resulted in the user can renumber the moves, so having gaps is not a problem. And both this way and setting the move to draft and repost it or recreating it does the same.
So I leave it like this until the forementioned task sees its end.